### PR TITLE
docs: new improved documentation [v2]

### PR DESCRIPTION
**Pull request by beraldoleal**
The current Avocado documentation is outdated and a bit confusing for
new users. Documentation is one of the starting points for understanding
how Avocado works and deciding whether to use it or not. This change
splits the current documentation into guides and chapters to improve the
readability and reduce the learning curve. During this process I tried
to change as minimum as possible the current content and added only a few
sections mainly on Contributor's Guide.

Signed-off-by: Beraldo Leal <bleal@redhat.com>

--------------------------------------

Changes from v1:

  * Removed `html:clean` from Makefile;
  * Removed "Code of Conduct";
  * Removed "Finally," on core's API header;
  * Few typo and suggestions fixed;
  * Moved built-in plugins to a proper chapter;
  * Fixed few broken references;
  * Bring back the APIs references;
  * Added two notes as discussed with Cleber;

### CODE REVIEW

File: docs/source/guides/user/chapters/plugins.rst [hunk 1]
```diff
@@ -3,6 +3,15 @@ Understanding the plugin system
 
 Avocado has a plugin system that can be used to extended it in a clean way.
 
+.. note:: A large number of out-of-the-box Avocado features are
+  implemented as using the same plugin architecture available to
+  third-party extensions.
+
+  This guide considers "core", although still plugable, features those
```
Comment (clebergnu) [inline docs/source/guides/user/chapters/plugins.rst]: Suggestion: "This guide considers "core features", even though they're still "plugable", those available with an installation of Avocado by itself..."

File: docs/source/guides/user/chapters/plugins.rst [hunk 2]
```diff
@@ -3,6 +3,15 @@ Understanding the plugin system
...
+  available with an installation of Avocado by itself (``pip install
+  avocado-framework``).  If a feature is part of an optional or
+  third-party plugin package, this guide will reference to it."
```
Comment (clebergnu) [inline docs/source/guides/user/chapters/plugins.rst]: "will reference it"

File: docs/source/guides/user/chapters/installing.rst [hunk 3]
```diff
@@ -11,6 +11,10 @@ repository, if available.
         Avocado features are distributed as non-core plugins. Visit the Avocado
         Plugin section on the left menu.
 
+.. tip:: If you are looking for Virtualization specific testing, also consider
```
Comment (clebergnu) [inline docs/source/guides/user/chapters/installing.rst]: Good!

File: docs/source/future/index.rst [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+   core/nrunner
+
+====================
+Experimental Plugins
+====================
+
+.. toctree::
+   :maxdepth: 1
+
+   plugins/cli.cmd.nrun
```
[the remark this answers is not in the document]
Comment (beraldoleal) [reply docs/source/future/index.rst]: Yes, the idea here was to have a proper documentation on each plugin per page.

File: docs/source/.include/helpus.rst [hunk 5]
```diff
@@ -0,0 +1 @@
+.. attention:: TODO: This section needs attention! Please, help us contributing on this document.
```
[the remark this answers is not in the document]
Comment (beraldoleal) [reply docs/source/.include/helpus.rst]: Not exactly.  Just a "convention" on my side. The idea was to have an (almost) 1:1 mapping between the guides/chapters with the directories, so on `ls -l` could be easy to find the things.

But I have no problem with changing this.

File: docs/source/guides/user/chapters/configuring.rst [hunk 6]
```diff
@@ -136,31 +144,38 @@ which uses ``*`` to mark existing and used files::
     datadir.paths.test_dir                   /usr/share/doc/avocado/tests
     ...
 
+Where the lower config files override values of the upper files and the
+``$HOME/.config/avocado/avocado.conf`` file missing.
```
[the remark this answers is not in the document]
Comment (beraldoleal) [reply docs/source/guides/user/chapters/configuring.rst]: Ok, I will do that.

File: docs/source/guides/writer/chapters/libs.rst [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+.. seealso:: If you would like a detailed API reference of this libraries,
+  please visit the "Reference API" section on the left menu.
+
+The following pages are the documentation for some of the Avocado utilities:
+
+.. warning:: TODO: Looks like the utils libraries documentation will be mainly
+  on docstrings, right?  If so, maybe makes sense to have only documented on API
+  reference? And any general instruction would be on module docstring. What you
+  guys think?
```
[the remark this answers is not in the document]
Comment (beraldoleal) [reply docs/source/guides/writer/chapters/libs.rst]: Yes, that is the idea. But I would suggest to keep this TODO and do this on a second phase. What you think?

[the remark this answers is not in the document]
Comment (clebergnu) [reply docs/source/guides/writer/chapters/libs.rst]: > Yes, that is the idea. But I would suggest to keep this TODO and do this on a second phase. What you think?

Works for me.

File: docs/source/.include/helpus.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (clebergnu) [reply docs/source/.include/helpus.rst]: > Not exactly. Just a "convention" on my side. The idea was to have an (almost) 1:1 mapping between the guides/chapters with the directories, so on `ls -l` could be easy to find the things.
> 
> But I have no problem with changing this.

OK, that's a valid reason (listing and seeing guides/chapters only).

File: docs/source/future/index.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (beraldoleal) [reply docs/source/future/index.rst]: @clebergnu are ok with this?

[the remark this answers is not in the document]
Comment (clebergnu) [reply docs/source/future/index.rst]: Yes, fine with me.

**Comment by beraldoleal** [comment]
I'm closing this PR in favor of a new version.

Thanks @clebergnu, for the comments and suggestions.

**Pull request closed without merging**